src/cca/scripts/cloud_socks_proxy_server.ts

```diff
@@ -1,15 +1,34 @@
+import * as cloud_social_provider from '../../lib/cloud/social/provider';
 import * as social from '../../interfaces/social';
 import * as uproxy_core_api from '../../interfaces/uproxy_core_api';
+import CoreConnector from '../../generic_ui/scripts/core_connector'
 
 import { SocksProxy } from '../model/socks_proxy_server';
 
+export function MakeCloudSocksProxy(corePromise: Promise<CoreConnector>,
```

Comment:
I agree. I am of the opinion that no object should need a "initialize" method. Initialization should be at construction time. This way you don't run into this kind of issue.

The problem we are discussing is due to the fact that core is overloaded with multiple responsibilities and needs to be initialized in multiple stages. However login and acceptInvitation should be completely unnecessary.